contracts/Minion.sol

```diff
@@ -30,6 +30,13 @@ contract Minion {
     function doWithdraw(address _token, uint256 _amount) public {
         moloch.withdrawBalance(_token, _amount);
     }
+    
+    function doCancel(uint256 _proposalId) public {
+        Action memory action = actions[_proposalId];
+        require(!action.executed, "Minion::action executed");
+        require(action.proposer == msg.sender, "Minion::Must be proposer");
+        moloch.cancelProposal(_proposalId);
+    }
```

Comment:
One note is that there's no modification of the proposal in the Minion state. I think we should remove the action, but maybe there's another reason we'd want to keep it around?

Also, an assumption we should clearly state is that Moloch does not allow a proposal that has already been sponsored to be canceled. There's no check on the status of the proposal here, so it just propagates a revert from Moloch if the proposal is sponsored

Reply:
we could add another propoerty to the struct of cancelled and clear out the data value. I don't think there is a reason to keep it.

on the second point. yeah, should we add a require against the sponsored and cancelled flags

```
        bool[6] memory flags = moloch.getProposalFlags(_proposalId);
        require(!flags[0], "Minion::proposal already sponsored");
        require(!flags[3], "Minion::proposal already cancelled");
```

Reply:
I'm still a proponent of deleting the cancelled action entirely, because otherwise we have proposals that are in an in-between state. As far as `minion.execute` is concerned, they are still valid proposals, but they have been cancelled in the moloch.

Also, I can add some tests for the cancel function.